microsoft/terminal
# Accessibility: Refactor IRenderData with IUiaData

## Summary of the Pull Request
Created the following inheritance tree:
`IBaseData`
├─ `IRenderData`
└─ `IUiaData`

This cleans up what is exposed to UiaProviders. If more things are needed in `IUiaData` that are already in `IRenderData`, just move them to `IBaseData`.

## References

## PR Checklist
* [x] Closes #1992 
* [x] ~Tests added/passed~ N/A
* [x] ~Requires documentation to be updated~ N/A
* [x] I am a core contributor

## Detailed Description of the Pull Request / Additional comments
Anything specifically used for the UIA Providers is now passed through IUiaData. I added a few `pragma region`s to make the code look pretty. No functionality should actually be changed here.

New files:
- `IBaseData`
- `IUiaData`

## Validation Steps Performed
Used inspect.exe to see if the UIA Tree still appears in ConHost and WindowsTerminal. I was able to still interact with the UIA Tree so nothing broke.
